### DISCUSSION TITLE
Istio upgrades are painful... because we have to keep updating kiali

### DISCUSSION THREAD

**Discussion by mikebell90** [kiali/kiali#6644] [Q&A]
Specifically these attributes
        config_map_name: "istio-1-16-4-20230724"
        istio_sidecar_injector_config_map_name: "istio-sidecar-injector-1-16-4-20230724"
        istiod_deployment_name: "istiod-1-16-4-20230724

Is there no way to make this dynamic, say with a label?

**Comment by jmazzitelli** [comment]
News to me that Istio users must update the Kiali CR due to specific version name and date tags in the resource names. Sounds like you have an odd installation mechanism? How do you install Istio?

**Comment by mikebell90** [comment]
We use the istio operator albeit circuituously using solo gloo reference to it using their crd. This then loads the istiooperator remotely and installs.https://docs.solo.io/gloo-mesh-enterprise/latest/setup/install/gloo_mesh_managed/

However doesn't any CANARY install do the same thing?

**Reply by mikebell90** [reply]
After all why are these configsble parameters if this wasn’t a general issue

**Reply by jmazzitelli** [reply]
It is likely because not many people use canary deployments of Istio. But yeah, I do remember this and it came up for canary upgrades.

We would welcome contributions to the Kiali project to eliminate the need to customize these in a canary deployment, if you are willing to take on that work. [Create a github enhancement request](https://github.com/kiali/kiali/issues/new?assignees=&labels=enhancement&projects=&template=feature_request.md&title=) where that work can be discussed.

**Reply by jmazzitelli** [reply]
I just noticed we already have a github issue created for this: https://github.com/kiali/kiali/issues/6043

**Reply by jmazzitelli** [reply]
HA, and I created one a while ago that is more relevant (I forgot about this one until I just saw it again):
* https://github.com/kiali/kiali/issues/6042

**Reply by jshaughn** [reply]
@jmazzitelli If we have two, please close one if it makes sense.

**Reply by jmazzitelli** [reply]
They are separate issues - one was to document the kiali behavior better, and the second was to see if we could make detection of the configmap better (or, at least have something in the code that returns the configmap name from a common function).
